### DISCUSSION TITLE
Haspmota on Lanbon L8 button events

### DISCUSSION THREAD

**Discussion by m3730r** [arendst/Tasmota#18642] [Support request] [answered]
Hi to all,

I'm using haspmota with a Lanbon L8, but I can't figure out, how to catch an event. 

My simple pages.jsonl:
{"page":0,"comment":"--- Page 0 ---"}
{"id":1,"obj":"btn","x":4,"y":4,"w":114,"h":101,"bg_color":"#000000","border_color":"Orange","border_width":2,"radius":10,"text":"Tag","text_color":"Orange","text_font":22}

The button is visible on the screen and also clickable.
Touch screen is recognized by haspmota
18:50:36.741 HTP: Berry Console
18:50:38.280 HTP: Consoles
18:50:39.430 HTP: Console
18:50:41.928 WIF: Checking connection...
18:50:53.504 TS : TSGlobal.touched x=60 y=66 (raw x=60 y=66)
18:50:53.606 TS : TSGlobal.touched x=60 y=66 (raw x=60 y=66)
18:50:53.702 TS : released x=60 y=66 (raw x=60 y=66)
18:50:54.301 TS : TSGlobal.touched x=43 y=53 (raw x=43 y=53)
18:50:54.416 TS : TSGlobal.touched x=43 y=53 (raw x=43 y=53)
18:50:54.497 TS : released x=43 y=53 (raw x=43 y=53)
18:50:55.105 TS : TSGlobal.touched x=56 y=63 (raw x=56 y=63)
18:50:55.141 BRY: GC from 16857 to 8944 bytes, objects freed 154/127 (in 1 ms) - slots from 391/440 to 132/350
18:50:55.201 TS : TSGlobal.touched x=56 y=63 (raw x=56 y=63)
18:50:55.300 TS : TSGlobal.touched x=56 y=63 (raw x=56 y=63)
18:50:55.397 TS : released x=56 y=63 (raw x=56 y=63)
18:50:56.001 TS : TSGlobal.touched x=61 y=58 (raw x=61 y=58)
18:50:56.116 TS : TSGlobal.touched x=61 y=58 (raw x=61 y=58)
18:50:56.205 TS : TSGlobal.touched x=61 y=58 (raw x=61 y=58)
18:50:56.301 TS : released x=61 y=58 (raw x=61 y=58)
18:50:56.904 TS : TSGlobal.touched x=57 y=59 (raw x=57 y=59)
18:50:57.021 TS : TSGlobal.touched x=57 y=59 (raw x=57 y=59)
18:50:57.101 TS : TSGlobal.touched x=57 y=59 (raw x=57 y=59)
18:50:57.197 TS : released x=57 y=59 (raw x=57 y=59)
18:50:57.803 TS : TSGlobal.touched x=51 y=56 (raw x=51 y=56)
18:50:57.903 TS : TSGlobal.touched x=51 y=56 (raw x=51 y=56)
18:50:58.000 TS : TSGlobal.touched x=51 y=56 (raw x=51 y=56)
18:50:58.098 TS : released x=51 y=56 (raw x=51 y=56)
18:51:01.926 WIF: Checking connection...

But shouldn't I see some events in the console too ? Is there something I have overseen?

Regards

**Comment by s-hadinger** [comment]
Pressing a button does generate internal events that are not shown in the console (or it would be bloated)

You can use rules to catch those events. Please see the documentation https://tasmota.github.io/docs/HASPmota/ at the bottom "React to user actions"

**Comment by m3730r** [comment]
Thank you for you hint: What are the events for a button? I've tried the following rules in the BE Console, like In the example, but nothing happens. 

Welcome to the Berry Scripting console. Check the [documentation](https://tasmota.github.io/docs/Berry/).
tasmota.add_rule("hasp#p0b1#event=up", / -> print("Button pressed"))
tasmota.add_rule("hasp#p0b1#event=pressed", / -> print("Button pressed"))

**Reply by sfromis** [reply, marked as answer]
You could try a debugging rule, to see what you get. Like
`tasmota.add_rule("hasp#p0b1#event", / value -> print('Event', value))`

More generally, do make sure to have upgraded to newest Tasmota version. There has been a bit of back and forth about how "rules" in Berry work around comparing values. While the docs currently explain that using double `==` is for numerical comparisons, and single `=` for string comparisons (matching what you do), at some recent point in time, Berry was doing it the other way.

**Reply by s-hadinger** [reply]
I tried your example and it works fine for me. Like sfromis mentioned, use the latest version

**Comment by m3730r** [comment]
The debugging rule works

tasmota.add_rule("hasp#p0b1#event", / value -> print('Event', value))
Event down
Event release
Event up

Thank you
